x-pack/plugins/enterprise_search/public/applications/app_search/components/meta_engine_creation/meta_engine_creation_logic.test.ts

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  LogicMounter,
+  mockHttpValues,
+  mockFlashMessageHelpers,
+  mockKibanaValues,
+} from '../../../__mocks__';
+
+import { nextTick } from '@kbn/test/jest';
+
+import { MetaEngineCreationLogic } from './meta_engine_creation_logic';
+
+describe('MetaEngineCreationLogic', () => {
+  const { mount } = new LogicMounter(MetaEngineCreationLogic);
+  const { http } = mockHttpValues;
+  const { navigateToUrl } = mockKibanaValues;
+  const { setQueuedSuccessMessage, flashAPIErrors } = mockFlashMessageHelpers;
+
+  const DEFAULT_VALUES = {
+    indexedEngineNames: [],
+    name: '',
+    rawName: '',
+    selectedIndexedEngineNames: [],
+  };
+
+  it('has expected default values', () => {
+    mount();
+    expect(MetaEngineCreationLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('setRawName', () => {
+      beforeAll(() => {
+        jest.clearAllMocks();
+        mount();
+        MetaEngineCreationLogic.actions.setRawName('Name__With#$&*%Special--Characters');
+      });
+
+      it('should set rawName to provided value', () => {
+        expect(MetaEngineCreationLogic.values.rawName).toEqual(
+          'Name__With#$&*%Special--Characters'
+        );
+      });
+
+      it('should set name to a sanitized value', () => {
+        expect(MetaEngineCreationLogic.values.name).toEqual('name-with-special-characters');
+      });
+    });
+
+    describe('setIndexedEngineNames', () => {
+      it('should set indexedEngineNames to provided value', () => {
+        mount();
+        MetaEngineCreationLogic.actions.setIndexedEngineNames(['first', 'middle', 'last']);
+        expect(MetaEngineCreationLogic.values.indexedEngineNames).toEqual([
+          'first',
+          'middle',
+          'last',
+        ]);
+      });
+    });
+
+    describe('setSelectedIndexedEngineNames', () => {
+      it('should set selectedIndexedEngineNames to provided value', () => {
+        mount();
+        MetaEngineCreationLogic.actions.setSelectedIndexedEngineNames(['one', 'two', 'three']);
+        expect(MetaEngineCreationLogic.values.selectedIndexedEngineNames).toEqual([
+          'one',
+          'two',
+          'three',
+        ]);
+      });
+    });
+  });
+
+  describe('listeners', () => {
+    describe('fetchIndexedEngineNames', () => {
+      beforeEach(() => {
+        jest.clearAllMocks();
+      });
+
+      it('calls flashAPIErrors on API Error', async () => {
+        http.get.mockReturnValueOnce(Promise.reject());
+        MetaEngineCreationLogic.actions.fetchIndexedEngineNames();
+        await nextTick();
+        expect(flashAPIErrors).toHaveBeenCalledTimes(1);
```

Comment:
at first I was a little confused why you put the API error catch test first, but I think I see now (based on the source code).

Super optional / up to you, but it might be nice to add an extra spy/assertion here to confirm that the code within `if (response)` didn't call, e.g.

```js
expect(MetaEngineCreationLogic.actions.setIndexedEngineNames).not.toHaveBeenCalled();
```

Reply:
Done!

Reply:
Not a blocker, but I don't see the above change/assertion in GH's diff.

As an aside, personally it would be super helpful for me if you included GH commit links to changes made as a comment response (even if there's multiple changes in a single commit), so I don't have to manually search for changes, but instead can click a link to immediately view the diff :)